### PR TITLE
Refactor legacy task routes

### LINKED ISSUE #5969: [BE] - Refactor legacy tasks endpoints into its own controller
1. create legacy_task_controller.rb with the following actions:
* create (judge case assignment to attorney)
* update (judge case reassignment to another attorney)

2. create case_review_controller with the following
* complete (attorney case review, judge case review)

3. leave tasks_controller.rb for new tasks (co-located, ama appeals, etc)
* leave index the way it is for now (will return legacy tasks for attorneys and judges)

### PR DESCRIPTION
**Pull request by aroltsch**
Fixes #5969 

1. create legacy_task_controller.rb with the following actions:
* create (judge case assignment to attorney)
* update (judge case reassignment to another attorney)

2. create case_review_controller with the following
* complete (attorney case review, judge case review)

3. leave tasks_controller.rb for new tasks (co-located, ama appeals, etc)
* leave index the way it is for now (will return legacy tasks for attorneys and judges)

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -138,14 +138,11 @@
     get '/appeals/:vacols_id', to: 'queue#index'
     get '/appeals/:vacols_id/*all', to: redirect('/queue/appeals/%{vacols_id}')
     get '/:user_id(*rest)', to: 'tasks#index'
-
-    post '/appeals/:id/complete', to: 'tasks#complete'
```
Comment (aroltsch) [inline config/routes.rb]: I am not seeing this route is used anywhere.

Comment (aroltsch) [reply config/routes.rb]: @lowellrex could you confirm please

File: app/controllers/issues_controller.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 class IssuesController < ApplicationController
-  before_action :verify_task_completion_access
+  before_action :verify_case_review_access
```
Comment (aroltsch) [inline app/controllers/issues_controller.rb]: Renamed the method to a better name

File: app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -1,21 +1,11 @@
 class TasksController < ApplicationController
   before_action :verify_queue_access
-  before_action :verify_task_completion_access, only: :complete
-  before_action :verify_task_assignment_access, only: [:create, :update]
-
-  rescue_from Caseflow::Error::VacolsRepositoryError do |e|
```
Comment (aroltsch) [inline app/controllers/tasks_controller.rb]: Moved this to the application controller

File: app/models/judge_case_assignment_to_attorney.rb [hunk 4]
```diff
@@ -2,7 +2,7 @@ class JudgeCaseAssignmentToAttorney
   include ActiveModel::Model
   include LegacyTaskConcern
 
-  attr_accessor :appeal_id, :assigned_to, :task_id, :assigned_by, :type
```
Comment (aroltsch) [inline app/models/judge_case_assignment_to_attorney.rb]: This class doesn't need `type` anymore.

File: config/routes.rb (hunk 1 above)
Comment (dannysellers) [reply config/routes.rb]: Confirmed!

**Pull request merged**